performance_manager/startup.py

```diff
@@ -0,0 +1,165 @@
+    """
+    try:
+        if int(os.environ.get("BOOTSTRAPPED", 0)) == 1:
+            return
+
+        env_file = os.path.join(HERE, "..", ".env")
+        logging.info("bootstrapping with env file %s", env_file)
+
+        with open(env_file, "r", encoding="utf8") as reader:
+            for line in reader.readlines():
+                line = line.rstrip("\n")
+                line.replace('"', "")
+                if line.startswith("#") or line == "":
+                    continue
+                key, value = line.split("=")
+                logging.info("setting %s to %s", key, value)
+                os.environ[key] = value
+
+    except Exception as exception:
+        logging.error("error while trying to bootstrap")
+        logging.exception(exception)
+        raise exception
+
+
+def parse_args(args: list[str]) -> argparse.Namespace:
+    """parse args for running this entrypoint script"""
+    parser = argparse.ArgumentParser(description=DESCRIPTION)
+    parser.add_argument(
+        "--experimental",
+        action="store_true",
+        dest="experimental",
+        help="if set, use a sqllite engine for quicker development",
+    )
+
+    parser.add_argument(
+        "--interval",
+        default=60,
+        dest="interval",
+        help="interval to run event loop on",
+    )
+
+    parser.add_argument(
+        "--seed",
+        action="store_true",
+        dest="seed",
+        help="if set, seed metadataLog table with paths",
+    )
+
+    return parser.parse_args(args)
+
+
+def seed_metadata_table(sql_session: sessionmaker) -> None:
+    """
+    add s3 filepaths to metadata table
+    TODO(team) make the filepath an input argument as well
+    """
+    try:
+        seed_file = os.path.join(HERE, "tests", "july_17_filepaths.json")
+        with open(seed_file, "r", encoding="utf8") as seed_json:
+            load_paths = json.load(seed_json)
+
+        with sql_session.begin() as session:  # type: ignore
+            session.execute(sa.insert(MetadataLog.__table__), load_paths)
+            session.commit()
+    except Exception as e:
+        logging.error("Error cleaning and seeding database")
+        logging.exception(e)
+
+
+def unprocessed_events(sql_session: sessionmaker) -> Dict[str, Dict[str, List]]:
+    """check metadata table for unprocessed parquet files"""
+    paths_to_load: Dict[str, Dict[str, List]] = {}
+    try:
+        read_md_log = sa.select((MetadataLog.pk_id, MetadataLog.path)).where(
+            MetadataLog.processed == sa.false()
+        )
+        with sql_session.begin() as session:  # type: ignore
+            for path_id, path in session.execute(read_md_log):
+                path = pathlib.Path(path)
+                if path.parent not in paths_to_load:
+                    paths_to_load[path.parent] = {"ids": [], "paths": []}
+                paths_to_load[path.parent]["ids"].append(path_id)
+                paths_to_load[path.parent]["paths"].append(str(path))
```

Comment:
I think we might want to move this into `lib/rt_vehicle_positions.py`. Each job will probably be performing actions on it's own segment of Metadata paths. So this probably doesn't need to be in the main event loop.

Reply:
i moved this to postgres utils. i think its going to end up returning all new files, not just the vehicle positions ones.